op-challenger/game/fault/register.go

```diff
@@ -58,10 +60,14 @@ func RegisterGameTypes(
 ) (CloseFunc, error) {
 	var closer CloseFunc
 	var l2Client *ethclient.Client
-	if cfg.TraceTypeEnabled(config.TraceTypeCannon) || cfg.TraceTypeEnabled(config.TraceTypePermissioned) {
-		l2, err := ethclient.DialContext(ctx, cfg.CannonL2)
+	if cfg.TraceTypeEnabled(config.TraceTypeCannon) || cfg.TraceTypeEnabled(config.TraceTypePermissioned) || cfg.TraceTypeEnabled(config.TraceTypeAsterisc) {
+		l2RpcUrl := cfg.CannonL2
+		if cfg.TraceTypeEnabled(config.TraceTypeAsterisc) {
+			l2RpcUrl = cfg.AsteriscL2
+		}
```

Comment:
I don't think this is right since we might have both cannon and asterisc enabled at the same time.  I don't really see a reason to support different L2s for cannon and asterisc so maybe we should just change `--cannon-l2` to be just `--l2` (with a `--cannon-l2` alias for backwards compatibility) and use it for both.